### DISCUSSION TITLE
best way to render large website

### DISCUSSION THREAD

**Discussion by jasonwjohns** [community/community#78068] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

We have an ever growing Github Pages site with some pages that take a significant amount of time to render. For this reason, we have been caching our pages so that they don't have to completely re-run every time we render the site. For example, when we add a new tab to the navigation bar, we need to re-render all pages so that the new tab shows up on each page. We have been utilizing the Large File Storage (LFS) system to store our cached files, but have gotten a couple of emails that we're nearing our capacity there.

Does anyone have advice for another way to efficiently update our site?

**Comment by erickdc7** [comment]
To efficiently update and render a large GitHub Pages site, consider the following strategies:

- **Lazy Loading**: Implement lazy loading to defer loading of non-critical resources at page load time. [This can significantly improve performance for pages with a lot of content](https://stackoverflow.com/questions/23578432/how-to-best-render-large-web-pages-of-mostly-tabular-content-5-mb-locally).
- **Optimize CSS**: Write efficient CSS by avoiding overly complex selectors and minimizing the number of DOM elements that need to be rendered.
- **Image Optimization**: Use appropriate file formats and compression techniques to reduce image sizes without compromising quality.
- **Use a CDN**: A Content Delivery Network (CDN) can cache your site's content and serve it from locations closer to your users, reducing load times.

These strategies can help you manage your site's growing content while keeping page load times in check. Remember to regularly review and optimize your site's assets and structure as it grows.
